packages/api/lambdas/index-from-database.js

```diff
@@ -98,11 +116,83 @@ async function indexModel({
   return totalItemsIndexed;
 }
 
+/**
+* indexModel - Index a postgres RDS table's contents to ElasticSearch
+*
+* @param {Object} params                  -- parameters
+* @param {any} params.esClient            -- ElasticSearch client
+* @param {any} params.postgresModel       -- @cumulus/db model
+* @param {string} params.esIndex          -- esIndex to write records to
+* @param {any} params.indexFn             -- Indexer function that maps to the database model
+* @param {any} params.limitEsRequests     -- limitEsRequests method (used for testing)
+* @param {Knex} params.knex               -- configured knex instance
+* @param {any} params.translationFunction -- function to translate postgres record
+*                                            to API record for ES
+* @param {number} params.pageSize         -- Page size for postgres pagination
+* @returns {number}                       -- number of items indexed
+*/
+async function indexModel({
+  esClient,
+  postgresModel,
+  esIndex,
+  indexFn,
+  limitEsRequests,
+  knex,
+  translationFunction,
+  pageSize,
+}) {
+  let startId = 1;
+  let totalItemsIndexed = 0;
+  let done;
+  let maxIndex = await postgresModel.getMaxCumulusId(knex);
+  /* eslint-disable no-await-in-loop */
+  while (done !== true && maxIndex > 0) {
+    const pageResults = await postgresModel.paginateByCumulusId(knex, startId, pageSize);
+    log.info(
+      `Attempting to index ${pageResults.length} records from ${postgresModel.tableName}`
+    );
+
+    const translatedResults = await Promise.all(
+      pageResults.map(async (result) => await translationFunction(result))
+    );
+
+    const indexPromises = translatedResults.map((result) => limitEsRequests(
+      async () => {
+        try {
+          return await indexFn(esClient, result, esIndex);
+        } catch (error) {
+          log.error(`Error indexing record ${JSON.stringify(result)}, error: ${error}`);
+          return false;
+        }
+      }
+    ));
+    const results = await Promise.all(indexPromises);
+    const successfulResults = results.filter((result) => result !== false);
+    totalItemsIndexed += successfulResults;
+
+    log.info(`Completed index of ${successfulResults.length} records from ${postgresModel.tableName}`);
+    startId += pageSize;
+    if (startId > maxIndex) {
+      startId = maxIndex;
+      log.info(`Updating maxIndex to account for new rows from ${postgresModel.tableName}`);
+      const oldMaxIndex = maxIndex;
+      maxIndex = await postgresModel.getMaxCumulusId(knex);
+      if (maxIndex <= oldMaxIndex) {
+        done = true;
+      }
+    }
+  }
+  /* eslint-enable no-await-in-loop */
+  return totalItemsIndexed;
+}
+
 async function indexFromDatabase(event) {
+  const knex = event.knex || getKnexClient();
   const {
     indexName: esIndex,
-    tables,
     esHost = process.env.ES_HOST,
+    dynamoTables = { reconciliationReportsTable: process.env.ReconciliationReportsTable },
```

Comment:
Why do we need this? Is this allowing us to set the env var for this table name when testing the lambda directly? Even though usually the lambda is invoked via an async operation from the API?

Reply:
Yes, that's how the dynamo tables were mapped in testing.   It's a holdover from the legacy method, which could be simplified given it's the *only* dynamo table.